{
    "cve_id": "CVE-2012-5339",
    "cve_description": "Multiple cross-site scripting (XSS) vulnerabilities in phpMyAdmin 3.5.x before 3.5.3 allow remote authenticated users to inject arbitrary web script or HTML via a crafted name of (1) an event, (2) a procedure, or (3) a trigger.",
    "cve_publish_date": "2012-10-25",
    "cwe_id": "CWE-79",
    "cwe_name": "Improper Neutralization of Input During Web Page Generation ('Cross-site Scripting')",
    "cwe_description": "The product does not neutralize or incorrectly neutralizes user-controllable input before it is placed in output that is used as a web page that is served to other users.",
    "commit_message": "triggers, routines, events : escape sql queries in error messages",
    "type_of_change": "Modification",
    "filename_of_changes": "rte_routines.lib.php",
    "code_language": "PHP",
    "number_of_lines_added_for_mitigation": "34",
    "number_of_lines_deleted_vulnerable_to_cve": "11",
    "vulnerable_lines": [
        "// Line_Reference 252:                         $errors[] = sprintf(__('The following query has failed: \"%s\"'), $drop_routine) . '<br />'",
        "// Line_Reference 253:                                           . __('MySQL said: ') . PMA_DBI_getError(null);",
        "// Line_Reference 257:                             $errors[] = sprintf(__('The following query has failed: \"%s\"'), $routine_query) . '<br />'",
        "// Line_Reference 258:                                               . __('MySQL said: ') . PMA_DBI_getError(null);",
        "// Line_Reference 266:                                 $errors[] = __('Sorry, we failed to restore the dropped routine.') . '<br />'",
        "// Line_Reference 267:                                                   . __('The backed up query was:') . \"\\\"$create_routine\\\"\" . '<br />'",
        "// Line_Reference 268:                                                   . __('MySQL said: ') . PMA_DBI_getError(null);",
        "// Line_Reference 281:                     $errors[] = sprintf(__('The following query has failed: \"%s\"'), $routine_query) . '<br /><br />'",
        "// Line_Reference 282:                                       . __('MySQL said: ') . PMA_DBI_getError(null);",
        "// Line_Reference 1254:                 $message = PMA_message::error(sprintf(__('The following query has failed: \"%s\"'), $query) . '<br /><br />'",
        "// Line_Reference 1255:                                                     . __('MySQL said: ') . PMA_DBI_getError(null));"
    ]
}
